networkx/algorithms/approximation/maxcut.py

```diff
@@ -0,0 +1,106 @@
+    corresponding cut value.
+
+    Parameters
+    ----------
+    G: NetworkX graph
+
+    seed : integer, random_state, or None (default)
+        Indicator of random number generation state.
+        See :ref:`Randomness<randomness>`.
+
+    p: double@nx.not_implemented_for('directed', 'multigraph')
+        Probability for each node to be part of the first partition.
+        Should be in [0,1]
+
+    weight : object
+        Edge attribute key to use as weight. If not specified, edges
+        have weight one.
+
+    Returns
+    -------
+    cut_size : integer, float
+        Value of the minimum cut.
+
+    partition : pair of node sets
+        A partitioning of the nodes that defines a minimum cut.
+    """
+    cut = set()
+    for node in G.nodes():
+        if seed.random() < p:
+            cut.add(node)
+    cut_size = nx.algorithms.cut_size(G, cut, weight=weight)
+    partition = (cut, set(G.nodes) - cut)
+    return cut_size, partition
+
+
+def _swap_node_partition(cut, node):
+    if node in cut:
+        new_cut = cut - {node}
+    else:
+        new_cut = cut.union({node})
+    return new_cut
+
+
+@nx.not_implemented_for('directed', 'multigraph')
+@py_random_state(2)
+def one_exchange(G, initial_cut=None, seed=None, weight=None):
+    """Compute a partitioning of the graphs nodes and the corresponding cut value.
+    Use a greedy one exchange strategy to find a locally maximal cut.
+
+    Parameters
+    ----------
+    G: networkx Graph
+        Graph to find a maximum cut for.
+
+    initial_cut: set
+        Cut to use as a starting point. If not supplied the algorithm
+        starts with an empty cut.
+
+    seed : integer, random_state, or None (default)
+        Indicator of random number generation state.
+        See :ref:`Randomness<randomness>`.
+
+    weight : object
+        Edge attribute key to use as weight. If not specified, edges
+        have weight one.
+
+    Returns
+    -------
+    cut_value : integer, float
+        Value of the maximum cut.
+
+    partition : pair of node sets
+        A partitioning of the nodes that defines a maximum cut.
+    """
+    if initial_cut is None:
+        initial_cut = set()
+    cut = set(initial_cut)
+    current_cut_size = nx.algorithms.cut_size(G, cut, weight=weight)
+    while True:
+        nodes = list(G.nodes())
+        seed.shuffle(nodes)
+        best_node_to_swap = max(nodes,
+                                key=lambda v:
+                                nx.algorithms.cut_size(G, _swap_node_partition(
+                                    cut, v), weight=weight), default=None)
+        potential_cut = _swap_node_partition(cut, best_node_to_swap)
+        potential_cut_size = nx.algorithms.cut_size(G, potential_cut, weight=weight)
+
+        if potential_cut_size > current_cut_size:
+            cut = potential_cut
+            current_cut_size = potential_cut_size
+        else:
+            break
+
+    partition = (cut, set(G.nodes) - cut)
```

Comment:
```set(G.nodes) - cut``` can be ```G.nodes - cut``` because we've made G.nodes and G.edges act as sets with the set operations.